File: frontend/src/api/apiClient.js
import axios from 'axios';

const rawBaseURL = import.meta.env.VITE_API_URL || 'http://localhost:3000';
const normalizedBaseURL = rawBaseURL.replace(/\/+$/, '');
const baseURL = normalizedBaseURL.endsWith('/api')
  ? normalizedBaseURL.slice(0, -4)
  : normalizedBaseURL;

const apiClient = axios.create({
  baseURL,
});

// Dynamically import Firebase auth and attach ID token to all requests
apiClient.interceptors.request.use(async (config) => {
  try {
    const { auth } = await import('../config/firebase');
    const user = auth.currentUser;
    if (user) {
      const idToken = await user.getIdToken();
      config.headers = config.headers || {};
      config.headers.Authorization = `Bearer ${idToken}`;
    }
  } catch (err) {
    // If Firebase not initialized or user not logged in, skip
  }
  return config;
}, (error) => Promise.reject(error));
apiClient.interceptors.response.use(
  (response) => response,
  (error) => {
    const message =
      error?.response?.data?.error ||
      error?.response?.data?.message ||
      error.message ||
      'Request failed';
    return Promise.reject(new Error(message));
  }
);

export default apiClient;

File: vendorerp/lib/core/api/api_client.dart
import 'package:dio/dio.dart';
import 'package:firebase_auth/firebase_auth.dart';

class ApiClient {
  final Dio dio = Dio(
    BaseOptions(baseUrl: "https://dqwltqhs-3000.inc1.devtunnels.ms/api"),
  );

  ApiClient() {
    dio.interceptors.add(
      InterceptorsWrapper(
        onRequest: (options, handler) async {
          final user = FirebaseAuth.instance.currentUser;
          if (user != null) {
            final token = await user.getIdToken();
            options.headers["Authorization"] = "Bearer $token";
          }
          return handler.next(options);
        },
      ),
    );
  }
}
